Packs/CheckPoint_FW/Integrations/CheckPoint_FW/CheckPoint_FW.py

```diff
@@ -0,0 +1,833 @@
+            application_signature(str): Application signature generated by Signature Tool
+        """
+
+        body = {'name': identifier,
+                'application-signature': application_signature,
+                'description': description,
+                'new-name': new_name,
+                'primary-category': primary_category,
+                'primary_category': primary_category,
+                'urls-defined-as-regular-expression': urls_defined_as_regular_expression,
+                }
+        response = self._http_request(method='POST', url_suffix='set-application-site',
+                                      headers=self.headers, json_data=body)
+        format_update_object(response, 'application-site')
+
+    def checkpoint_delete_application_site_command(self, identifier: str):
+        """
+        Delete existing application site object using object name or uid.
+
+        Args:
+            identifier(str): uid or name.
+        """
+        response = self._http_request(method='POST', url_suffix='delete-application-site',
+                                      headers=self.headers, json_data={'name': identifier})
+        format_delete_object(response, 'application-site')
+
+    def checkpoint_publish_command(self):
+        """
+        publish changes.
+        All the changes done by this user will be seen by all users only after publish is called.
+        """
+        return self._http_request(method='POST', url_suffix='publish', headers=self.headers,
+                                  json_data={})
+
+    def install_policy_command(self, policy_package: str, targets, access: bool):
+        """
+        installing policy.
+
+        Args:
+            policy_package(str): The name of the Policy Package to be installed.
+            targets(str or list):On what targets to execute this command. Targets may be identified
+                                by their name, or object unique identifier.
+            access(bool): Set to be true in order to install the Access Control policy.
+                            By default, the value is true if Access Control policy is enabled
+                            on the input policy package, otherwise false.
+        """
+        body = {
+            'policy-package': policy_package,
+            'targets': targets,
+            'access': access
+        }
+        return self._http_request(method='POST', url_suffix='install-policy',
+                                  headers=self.headers, json_data=body)
+
+
+def login(base_url: str, username: str, password: str, verify_certificate: bool = False) -> str:
+    """
+    login to checkpoint admin account using username and password.
+    """
+    return requests.post(base_url + 'login', verify=verify_certificate,
+                         headers={'Content-Type': 'application/json'},
+                         json={'user': username, 'password': password}).json().get('sid')
```

Comment:
why is this not a method of the client object?

Reply:
Login returns session ID (sid). This sid is part of the headers and the headers are object property.